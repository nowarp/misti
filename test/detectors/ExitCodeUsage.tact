 contract C {
     owner: Address;
     init() { self.owner = sender(); }

     fun testDivByZero(unknown: Int) {
         // OK: No exception
         let a: Int = 128;
         let b: Int = a / unknown;
         nativeThrowUnless(b, sender() == self.owner);
     }

     receive("test") {
         // Bad
         let code1: Int = 128;
         nativeThrowUnless(code1, sender() == self.owner);

         // Bad
         let code2: Int = 256;
         code2 = code2 - 10;
         nativeThrowUnless(code2, sender() == self.owner);

         // OK: No false positive
         let code3: Int = 257;
         nativeThrowUnless(code3, sender() == self.owner);
     }
 }
