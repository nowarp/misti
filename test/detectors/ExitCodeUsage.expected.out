[HIGH] ExitCodeUsage: Exit code variable "code1" has value outside allowed range
test/detectors/ExitCodeUsage.tact:15:28:
  14 |          let code1: Int = 128;
> 15 |          nativeThrowUnless(code1, sender() == self.owner);
                                  ^
  16 | 
Exit codes 0-255 are reserved. Variable value: 128
Help: Use a value between 256 and 65535
See: https://nowarp.io/tools/misti/docs/detectors/ExitCodeUsage

[HIGH] ExitCodeUsage: Exit code variable "code2" has value outside allowed range
test/detectors/ExitCodeUsage.tact:20:28:
  19 |          code2 = code2 - 10;
> 20 |          nativeThrowUnless(code2, sender() == self.owner);
                                  ^
  21 | 
Exit codes 0-255 are reserved. Variable value: 246
Help: Use a value between 256 and 65535
See: https://nowarp.io/tools/misti/docs/detectors/ExitCodeUsage